### ISSUE TITLE
code comment

### ISSUE THREAD

**Issue by Linshiqi-Git** [GuillaumeSalhaGalvan/modularity_aware_gae#1]
Hey,I noticed that there no item in the loss function that adds whether the node belongs to a community, as mentioned in the article δ(i, j).
Is there something I missed?

Best regard,
Lin

**Comment by GuillaumeSalhaGalvan** [comment]
Dear @EEEleanor,

First of all, thank you for your message and interest in our work! :) 

I assume you are referring to the δ(i, j) term from equation (15) of page 16 [of the paper](https://arxiv.org/pdf/2202.00961.pdf), which corresponds to the "exact" modularity term. 

The loss functions of our Modularity-Aware GAE/VGAE do not involve this "exact" modularity. Instead, we leverage the "modularity-inspired" terms defined in equations (18) and (19) of page 20. As you correctly noticed, they are independent of the community information associated with nodes. 

More specifically, equations (18) and (19) involve an exponentialized L2 distance term between embedding vectors of nodes i and j (taking values in [0, 1]) that acts as a soft counterpart of the community indicator δ(i, j) ∈ {0, 1} of equation (15). It tends to 1 when nodes i and j get closer in the embedding space, and tends to 0 when they move apart.

We favored the use of this "modularity-inspired" formulation over the "exact" one, as it permits to obtain a differentiable loss, and also to avoid the actual reconstruction of node communities at each training iteration (e.g., by running a k-means in the embedding space), which would incur a larger computational expense.

In the code, these exponentialized L2 distance terms are computed in the `DistanceDecoder` of `layer.py`, and correspond to the `clusters_distance` arguments from functions in `optimizer.py`.

Best regards,

Guillaume
